### PR TITLE
feat(pipeline): leader-worekr safe invoke functions

### PR DESCRIPTION
**Pull request by sfwn**
#### What this PR does / why we need it:

leader-worekr safe invoke functions

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  ppipeline leader-worekr provider safe invoke functions            |
| 🇨🇳 中文    |  流水线 leader-worker 模块安全调用函数            |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/2.1-beta.3` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/providers/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -18,15 +18,20 @@ import (
...
 
 func (p *provider) continueDispatcher(ctx context.Context) {
-	for pipelineID := range p.pipelineIDsChan {
-		go func(pipelineID uint64) {
-			p.dispatchOnePipelineUntilSuccess(ctx, pipelineID)
-		}(pipelineID)
+	for {
+		select {
+		case <-ctx.Done():
+			p.Log.Infof("stop continue dispatcher, reason: %v", ctx.Done())
```
Comment (chengjoey) [inline modules/pipeline/providers/dispatcher/dispatcher.go:30]: why ctx.Done()

Comment (sfwn) [reply modules/pipeline/providers/dispatcher/dispatcher.go:30]: such as leader changed.

Comment (sfwn) [reply modules/pipeline/providers/dispatcher/dispatcher.go:30]: if ctx done, dispatcher should stop `continueDispath`.

Comment (sfwn) [reply modules/pipeline/providers/dispatcher/dispatcher.go:30]: changed to `ctx.Err()`

**Review by chengjoey — APPROVED** [review]

**Pull request merged**